iOS/Source/MainController.swift

```diff
@@ -1,121 +1,236 @@
-            alertController.addAction(UIAlertAction(title: "Dismiss", style: .default, handler: nil))
-            self.present(alertController, animated: true, completion: nil)
-        }
-    }
+        let y = self.messageLabel.frame.maxY - height + margin
+
+        let overlayView = UIView(frame: CGRect(x: 0, y: y, width: self.view.frame.width, height: height))
+        self.view.addSubview(overlayView)
+
+        self.informationButtonTopAnchor?.isActive = false
+        let screenshotInformationTopAnchor = self.informationButton.top(to: overlayView, offset: -18)
+
+        self.locationLabelBottomAnchor?.isActive = false
+        self.locationLabelLeftAnchor?.isActive = false
+        self.locationLabelRightAnchor?.isActive = false
+
+        let screenshotLocationCenterYAnchor = self.locationLabel.topAnchor.constraint(equalTo: self.informationButton.centerYAnchor)
+        screenshotLocationCenterYAnchor.isActive = true
+
+        let screenshotLocationRightAnchor = self.locationLabel.right(to: self.view, offset: -screenShotInsets.right)
+        let screenshotLocationLeftAnchor = self.locationLabel.left(to: self.view, offset: daylightLabelWidth + screenShotInsets.right)
+
+        self.locationLabel.textAlignment = .right
 
-    func locationTracker(_ locationTracker: LocationTracker, didFindLocation placemark: CLPlacemark) {
-        Location.current = Location(placemark: placemark)
+        self.informationButton.nameLabel.text = NSLocalizedString("Daylight", comment: "")
+        self.informationButton.nameLabel.sizeToFit()
 
-        self.rootView.updateInterface(location: Location.current)
+        let screenshot = UIScreen.screenshot(frame: overlayView.frame)
+
+        self.informationButton.nameLabel.text = NSLocalizedString("About", comment: "")
+
+        self.locationLabel.textAlignment = .left
+        screenshotLocationLeftAnchor.isActive = false
+        screenshotLocationRightAnchor.isActive = false
+        screenshotLocationCenterYAnchor.isActive = false
+
+        self.locationLabelBottomAnchor?.isActive = true
+        self.locationLabelLeftAnchor?.isActive = true
+        self.locationLabelRightAnchor?.isActive = true
+
+        screenshotInformationTopAnchor.isActive = false
+        self.informationButtonTopAnchor?.isActive = true
+
+        overlayView.removeFromSuperview()
+
+        return screenshot
     }
 }
 
 extension MainController: InformationControllerDelegate {
 
-    func informationController(_ informationController: InformationController, didToggleNotifications isNotificationsEnabled: Bool) {
-        if isNotificationsEnabled {
-            if let location = Location.current {
-                Notifier.scheduleNotifications(for: location)
-            }
-        } else {
-            Notifier.cancelAllNotifications()
+    func didToggleNotifications(_ isNotificationsEnabled: Bool, on informationController: InformationController) {
+        //maybe make something else responsible for this
```

Comment:
a) TODO plz
b) What do you think should be responsible for this?

Reply:
Im thinking the InitialViewController, but actually not sure since it might get too big... And its not really related to app setup or anything.. Don't know yet!